### PR TITLE
Use Xcode 14

### PR DESCRIPTION
**Pull request by jmpg93**
Hi there!
This PR is about making the repo to start using Xcode 14.
I've updated the simulator to use the latests one for testing, it simplifies a lot simulator management as iOS 16 is included with Xcode 14.

- There are a lot of files because snapshots has been updated. 
- I've manually checked the screenshots one by one.

The real changes are those files (everything that is no under the test folder):
image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,17 +37,27 @@ help:
...
 
-setup:
+trace:
+	@echo "Current xcodebuild configuration"
+	@xcodebuild -version
+	@echo "Available schemas"
+	@xcodebuild -list -json
+	@echo "Available devices"
+	@xcrun xctrace list devices
+	@xcrun --sdk iphonesimulator --show-sdk-path
```
Comment (jmpg93) [inline Makefile:47]: The current mistica SPM configuration is problematic and unestable. So I'm adding this make lane to be able to get more info when the job fail.

Comment (jmpg93) [inline Makefile:47]: This job is really unstable that's why I've added this trace and I've splitted the testing phase in two.

File: Makefile [hunk 2]
```diff
@@ -37,17 +37,27 @@ help:
...
 format:
-	Scripts/swiftformat . 
-
+	Scripts/swiftformat .
+          
 test: clean setup simulator
 	@echo "Testing with simulator $(SIMULATOR_NAME)"
-	$(XCODEBUILD) -scheme $(TEST_SCHEMA) test -sdk iphonesimulator -resultBundlePath $(XCRESULT_FILE_PATH) -destination 'platform=iOS Simulator,name=$(SIMULATOR_NAME)' | xcbeautify
+	$(XCODEBUILD) build-for-testing -scheme $(TEST_SCHEMA) -destination "platform=iOS Simulator,name=$(SIMULATOR_NAME)" | xcbeautify
+	$(XCODEBUILD) test-without-building -scheme $(TEST_SCHEMA) -resultBundlePath $(XCRESULT_FILE_PATH) -destination "platform=iOS Simulator,name=$(SIMULATOR_NAME)" | xcbeautify
```
Comment (jmpg93) [inline Makefile:60]: With the current SPM configuration xcodebuilds breaks sometimes with a Segmentation fault 11: https://github.com/Telefonica/mistica-ios/actions/runs/3098227241/jobs/5015964296

I've splitted this in two to have more visibility about the failing step.

**Review by idenjoe — APPROVED** [review]

**Review by cbarcom — APPROVED** [review]

**Pull request merged**